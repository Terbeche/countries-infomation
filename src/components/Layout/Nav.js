import '../../css/Navigation.css';
import { useSelector, useDispatch } from 'react-redux';
import { NavLink, Link } from 'react-router-dom';
import React from 'react';
import { FiSettings } from 'react-icons/fi';
import { GrMicrophone } from 'react-icons/gr';
import { filterCountries, fetchCountries } from '../../redux/countries/countries';

const Navigation = () => {
  const dispatch = useDispatch();
  const { Countries } = useSelector((state) => state.countries);

  const handleContinentChoice = () => {
    if (document.getElementById('myDropdown').classList.toggle === 'show') {
      document.getElementById('myDropdown').classList.toggle('hide');
    } else {
      document.getElementById('myDropdown').classList.toggle('show');
    }
  };

  const helperFunction = () => {
    dispatch(fetchCountries());
  };

  const handleContinentSelection = (continentArg) => {
    if (document.getElementById('myDropdown').classList.toggle === 'show') {
      document.getElementById('myDropdown').classList.toggle('hide');
    } else {
      document.getElementById('myDropdown').classList.toggle('show');
    }
    const countryDetail = Countries.filter(({ continent }) => continent === continentArg);
    dispatch(filterCountries(countryDetail));
  };

  return (
    <header>
      <nav className="nav">
        <div className="logo Flex-1">
          <Link to="/">
            <h2>Countries Hub</h2>
          </Link>
        </div>
        <div className="dropdown">
          <button type="button" onClick={() => { helperFunction(); handleContinentChoice(); }} className="dropbtn">Continents</button>
          <div id="myDropdown" className="dropdown-content">
            <button type="button" className="dropbtn" onClick={() => handleContinentSelection('Africa')}>Africa</button>
            <button type="button" className="dropbtn" onClick={() => handleContinentSelection('Asia')}>Asia</button>
            <button type="button" className="dropbtn" onClick={() => handleContinentSelection('Europe')}>Europe</button>
            <button type="button" className="dropbtn" onClick={() => handleContinentSelection('North America')}>North America</button>
            <button type="button" className="dropbtn" onClick={() => handleContinentSelection('South America')}>South America</button>
            <button type="button" className="dropbtn" onClick={() => handleContinentSelection('Oceania')}>Oceania</button>
            <button type="button" className="dropbtn" onClick={() => handleContinentSelection('Antarctica')}>Antarctica</button>

          </div>
        </div>

        <ul className="navLinks Flex-1">
          <li className="linkItems">
            <NavLink
              to="countries"
            >
              <GrMicrophone />
            </NavLink>
            <NavLink
              to="countries"
            >
              <FiSettings />
            </NavLink>
          </li>
        </ul>
      </nav>
    </header>

  );
};

export default Navigation;
