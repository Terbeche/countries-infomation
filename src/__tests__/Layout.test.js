import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Layout from '../components/Layout/Layout';

it('Layout component snapshot testing', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Layout />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
