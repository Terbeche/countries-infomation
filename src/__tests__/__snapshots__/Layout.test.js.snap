// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component snapshot testing 1`] = `
<div>
  <header>
    <nav
      className="nav"
    >
      <div
        className="logo Flex-1"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <h2
            className="logo-name"
          >
            Countries Info
          </h2>
        </a>
      </div>
      <div
        className="dropdown"
      >
        <button
          className="dropbtn"
          onClick={[Function]}
          type="button"
        >
          Continents
        </button>
        <div
          className="dropdown-content"
          id="myDropdown"
        >
          <button
            className="dropbtn"
            onClick={[Function]}
            type="button"
          >
            Africa
          </button>
          <button
            className="dropbtn"
            onClick={[Function]}
            type="button"
          >
            Asia
          </button>
          <button
            className="dropbtn"
            onClick={[Function]}
            type="button"
          >
            Europe
          </button>
          <button
            className="dropbtn"
            onClick={[Function]}
            type="button"
          >
            North America
          </button>
          <button
            className="dropbtn"
            onClick={[Function]}
            type="button"
          >
            South America
          </button>
          <button
            className="dropbtn"
            onClick={[Function]}
            type="button"
          >
            Oceania
          </button>
          <button
            className="dropbtn"
            onClick={[Function]}
            type="button"
          >
            Antarctica
          </button>
        </div>
      </div>
      <ul
        className="navLinks Flex-1"
      >
        <li
          className="linkItems"
        >
          <a
            className="single-item"
            href="/countries"
            onClick={[Function]}
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 3a3 3 0 0 1 6 0v5a3 3 0 0 1-6 0V3z"
              />
              <path
                d="M3.5 6.5A.5.5 0 0 1 4 7v1a4 4 0 0 0 8 0V7a.5.5 0 0 1 1 0v1a5 5 0 0 1-4.5 4.975V15h3a.5.5 0 0 1 0 1h-7a.5.5 0 0 1 0-1h3v-2.025A5 5 0 0 1 3 8V7a.5.5 0 0 1 .5-.5z"
              />
            </svg>
          </a>
        </li>
        <li
          className="linkItems"
        >
          <a
            className="single-item"
            href="/countries"
            onClick={[Function]}
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="3"
              />
              <path
                d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
              />
            </svg>
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    className="main"
  />
</div>
`;
